sympy/integrals/transforms.py

```diff
@@ -1005,14 +1007,45 @@ def repl(ex, *args):
     expr = repl(expr, Unequality, replue)
     return S(expr)
 
+def expand_dirac_delta(expr):
+    """
+    Expand an expression involving DiractDelta to get it as a linear
+    combination of DiracDelta functions.
+    """
+    return _lin_eq2dict(expr, expr.atoms(DiracDelta))
 
 @_noconds
 def _laplace_transform(f, t, s_, simplify=True):
     """ The backend function for Laplace transforms. """
     from sympy import (re, Max, exp, pi, Min, periodic_argument as arg_,
-                       arg, cos, Wild, symbols, polar_lift)
+                       arg, cos, Wild, symbols, polar_lift, Add)
     s = Dummy('s')
-    F = integrate(exp(-s*t) * f, (t, 0, oo))
+    a = Wild('a', exclude=[t])
+    deltazero = 0
+    try:
+        integratable, deltadict = expand_dirac_delta(f)
+    except PolyNonlinearError:
+        raise IntegralTransformError(
+        'Laplace', f, 'could not expand DiracDelta expressions')
+    for dd in deltadict.keys():
```

Comment:
It is never necessary in Python to iterate over `d.keys` since iterating directly over `d` is the same i.e.:
```python
for dd in deltadict:
```
Actually though if you also want the corresponding value (which you do below) then it is better to iterate over items:
```python
for dirac_func, dirac_coeff in deltadict.items():
```

Reply:
OK!